src/ansiblelint/config.py

```
@@ -315,11 +315,11 @@ def _fetch_latest_release(cache_file: str) -> dict[str, Any]:
         msg = "release_url must start with https://"
         raise ValueError(msg)
     try:
-        with urllib.request.urlopen(release_url) as url:
+        with urllib.request.urlopen(release_url, timeout=10) as url:
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)